src/video_core/rasterizer_cache/custom_tex_manager.cpp

```diff
@@ -0,0 +1,395 @@
+        LoadTexture(texture);
+        return upload();
+    }
+
+    // Don't submit a decode multiple times.
+    if (texture->IsNone()) {
+        texture->state = DecodeState::Pending;
+        workers->QueueWork([this, texture] { LoadTexture(texture); });
+    }
+
+    // Queue the upload for later processing.
+    async_uploads.push_back({
+        .texture = texture,
+        .func = std::move(upload),
+    });
+
+    return false;
+}
+
+void CustomTexManager::LoadTexture(CustomTexture* texture) {
+    if (texture->IsDecoded()) {
+        return;
+    }
+
+    MICROPROFILE_SCOPE(CustomTexManager_ComputeHash);
+
+    FileUtil::IOFile file{texture->path, "rb"};
+    const size_t read_size = file.GetSize();
+    std::vector<u8> in(read_size);
+    if (file.ReadBytes(in.data(), read_size) != read_size) {
+        LOG_CRITICAL(Render, "Failed to open {}", texture->path);
+    }
+
+    switch (texture->file_format) {
+    case CustomFileFormat::PNG: {
+        if (!image_interface.DecodePNG(in, texture->data, texture->width, texture->height)) {
+            LOG_ERROR(Render, "Failed to decode png {}", texture->path);
+        }
+        if (flip_png_files) {
+            Common::FlipRGBA8Texture(texture->data, texture->width, texture->height);
+        }
+        texture->format = CustomPixelFormat::RGBA8;
+        break;
+    }
+    case CustomFileFormat::DDS:
+    case CustomFileFormat::KTX: {
+        // Compressed formats don't need CPU decoding and must be pre-flipped.
+        ddsktx_format format{};
+        image_interface.DecodeDDS(in, texture->data, texture->width, texture->height, format);
+        texture->format = ToCustomPixelFormat(format);
+        break;
+    }
+    default:
+        LOG_ERROR(Render, "Unknown file format {}", texture->file_format);
+        return;
+    }
+    texture->state = DecodeState::Decoded;
+}
+
+void CustomTexManager::ReadConfig(const std::string& load_path) {
+    const std::string config_path = load_path + "pack.ini";
+    Common::INIReader reader{config_path};
+    if (!reader.IsOpen()) {
+        LOG_INFO(Render, "Unable to find pack config file, using legacy defaults");
+        refuse_dds = true;
+        return;
+    }
+
+    // Read config options
+    skip_mipmap = reader.GetBoolean("Options", "skip_mipmap", false);
+    if (skip_mipmap) {
+        LOG_WARNING(Render, "Skip mipmap option is enabled, pack is considered legacy!");
+        refuse_dds = true;
+    }
+    flip_png_files = reader.GetBoolean("Options", "flip_png_files", false);
+    use_new_hash = reader.GetBoolean("Options", "use_new_hash", true);
+    if (!use_new_hash) {
+        LOG_WARNING(Render, "Legacy hash is used, pack is considered legacy!");
+        refuse_dds = true;
+    }
+
+    // Read any hash mappings
+    auto& hashes = reader.Get("Hashes");
+    for (const auto& [key, file] : hashes) {
+        size_t idx{};
+        const unsigned long long hash = std::stoull(key, &idx, 16);
```

Comment:
```suggestion
        const u64 hash = std::stoull(key, &idx, 16);
```
same with all other occurences